fdbserver/workloads/MetaclusterManagementWorkload.actor.cpp

```diff
@@ -388,12 +388,25 @@ struct MetaclusterManagementWorkload : TestWorkload {
 		state bool exists = itr != self->createdTenants.end();
 		state bool hasCapacity = self->createdTenants.size() < self->totalTenantGroupCapacity;
 		state bool retried = false;
+		state bool preferAssignedCluster = deterministicRandom()->coinflip();
 
 		try {
 			loop {
 				try {
-					Future<Void> createFuture =
-					    MetaclusterAPI::createTenant(self->managementDb, tenant, TenantMapEntry());
+					TenantMapEntry entry;
+					if (preferAssignedCluster) {
+						loop {
+							ClusterName clusterName = self->chooseClusterName();
+							DataClusterData* dataDb = &self->dataDbs[clusterName];
+							// Choose a registered cluster
+							// Still possible that the chosen cluster has no capacity.
+							if (dataDb->registered) {
```

Comment:
I can't remember, is it possible that there are no registered clusters and we could get stuck here? Perhaps we could just use whatever cluster is chosen regardless of whether it is registered and then make sure we get the appropriate error if it isn't.